plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerDelegate.java

```diff
@@ -1,37 +1,105 @@
 package com.google.cloud.tools.eclipse.appengine.localserver.server;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.NullProgressMonitor;
 import org.eclipse.core.runtime.Status;
+import org.eclipse.jst.server.core.IWebModule;
 import org.eclipse.wst.server.core.IModule;
+import org.eclipse.wst.server.core.IModuleType;
+import org.eclipse.wst.server.core.IServer;
+import org.eclipse.wst.server.core.internal.facets.FacetUtil;
 import org.eclipse.wst.server.core.model.ServerDelegate;
 
+@SuppressWarnings("restriction") // For FacetUtil
 public class LocalAppEngineServerDelegate extends ServerDelegate {
+  private static final String SERVLET_MODULE_FACET = "jst.web";
+  private static final String ATTR_CLOUD_SDK_SERVER_MODULES = "cloudsdk-server-modules-list";
+	
+  public static LocalAppEngineServerDelegate getAppEngineServer(IServer server) {
+	LocalAppEngineServerDelegate appEngineServer = server.getAdapter(LocalAppEngineServerDelegate.class);
+    if (appEngineServer == null) {
+    	appEngineServer = (LocalAppEngineServerDelegate) server.loadAdapter(LocalAppEngineServerDelegate.class,
+                                                           new NullProgressMonitor());
```

Comment:
ProgressMonitors, if I understand them, are for reporting completion percentage to the user, not for reporting success/failure. Even if it's guaranteed to succeed, we still want to report long-running progress to the end user.